cartography/intel/aws/inspector.py

```diff
@@ -193,41 +290,73 @@ def _load_findings_tx(
-        ON CREATE SET r.firstseen = timestamp()
-        SET r.lastupdated = $UpdateTag
+        MERGE (account)-[r0:RESOURCE]->(finding)
+        ON CREATE SET r0.firstseen = timestamp()
+        SET r0.lastupdated = $UpdateTag
         WITH finding
-        MATCH (instance:EC2Instance{id: finding.instanceid})
-        MERGE (instance)<-[r2:AFFECTS]-(finding)
+        MATCH (package:AWSInspectorPackage) where finding.vulnerablepackageids CONTAINS package.id
+        MERGE (finding)-[r1:AFFECTS]->(package)
+        SET r1.lastupdated = $UpdateTag
+    """
+
+    query2 = """
+    UNWIND $Findings as new_finding
+        MATCH (finding:AWSInspectorFinding{id: new_finding.id})
+        WITH finding
+        MATCH (lambda:AWSLambda) where finding.lambdaid = lambda.id
+        MERGE (lambda)<-[r2:PRESENT_ON]-(finding)
         ON CREATE SET r2.firstseen = timestamp()
         SET r2.lastupdated = $UpdateTag
+    """
+
+    query3 = """
+    UNWIND $Findings as new_finding
+        MATCH (finding:AWSInspectorFinding{id: new_finding.id})
         WITH finding
-        MATCH (repo:ECRRepository{id: finding.ecrrepositoryid})
-        MERGE (repo)<-[r3:AFFECTS]-(finding)
+        MATCH (instance:EC2Instance) where finding.instanceid = instance.id
+        MERGE (instance)<-[r3:PRESENT_ON]-(finding)
         ON CREATE SET r3.firstseen = timestamp()
         SET r3.lastupdated = $UpdateTag
+    """
+
+    query4 = """
+    UNWIND $Findings as new_finding
+        MATCH (finding:AWSInspectorFinding{id: new_finding.id})
         WITH finding
-        MATCH (image:ECRImage{id: finding.ecrimageid})
-        MERGE (image)<-[r4:AFFECTS]-(finding)
+        MATCH (repo:ECRRepository{id: finding.ecrrepositoryid})
+        MERGE (repo)<-[r4:PRESENT_ON]-(finding)
         ON CREATE SET r4.firstseen = timestamp()
         SET r4.lastupdated = $UpdateTag
     """
 
-    tx.run(
-        query,
-        Findings=findings,
-        UpdateTag=aws_update_tag,
-        Region=region,
-    )
+    query5 = """
+    UNWIND $Findings as new_finding
+        MATCH (finding:AWSInspectorFinding{id: new_finding.id})
+        WITH finding
+        MATCH (image:ECRImage{id: finding.ecrimageid})
+        MERGE (image)<-[r5:PRESENT_ON]-(finding)
+        ON CREATE SET r5.firstseen = timestamp()
+        SET r5.lastupdated = $UpdateTag
+    """
+    queries = [query, query2, query3, query4, query5]
+    for query in queries:
+        tx.run(
+            query,
+            Findings=findings,
+            UpdateTag=aws_update_tag,
+            Region=region,
+        )
 
 
 @timeit
 def load_inspector_findings(
-    neo4j_session: neo4j.Session, findings: List[Dict], region: str,
+    neo4j_session: neo4j.Session,
+    findings: List[Dict],
+    region: str,
     aws_update_tag: int,
 ) -> None:
     for i, findings_batch in enumerate(batch(findings), start=1):
-        logger.info(f'Loading batch number {i}')
+        logger.info(f"Loading batch number {i}")
```

Comment:
We might need in the future to load multiple batches asynchronously, since this job can take some time to finish.